### ISSUE TITLE
`no-top-level-side-effects`: reports Vue SFC top-level calls to `watch*`, `define*`, etc.

### ISSUE THREAD

**Issue by andreww2012** [sindresorhus/eslint-plugin-unicorn#3538]
I'm not sure how this can be fixed, but currently the rule reports basic Vue building blocks like `watch`, `watchEffect`, `define*` calls at top level, as they return something and (formally correctly) are reported as top-level side effects:

```vue
<script setup>
import {watch} from 'vue';

watch(...); // Reported!
</script>
```

However, this basically makes this rule unusable in this scenario, despite it still being potentially useful.

Maybe a sort of functions whitelist should be allowed to be specified in rule options? Maybe prefixing calls with `void` should cause the rule not to report?
